simulators/optimism/l1ops/deposit_tests.go

```diff
@@ -270,13 +272,75 @@ func erc20RoundtripTest(t *hivesim.T, env *optimism.TestEnv) {
 	require.EqualValues(t, big.NewInt(500), balL2)
 }
 
-func doDeposit(t *hivesim.T, env *optimism.TestEnv, depositor common.Address, mintAmount *big.Int) {
-	depositContract, err := bindings.NewOptimismPortal(
-		env.Devnet.Deployments.DeploymentsL1.OptimismPortalProxy,
-		env.Devnet.L1Client(0),
-	)
+func failingDepositWithMintTest(t *hivesim.T, env *optimism.TestEnv) {
+	// Initial setup
+	l1 := env.Devnet.L1Client(0)
+	l2 := env.Devnet.L2Client(0)
+	l1Vault := env.Devnet.L1Vault
+	l2Vault := env.Devnet.L2Vault
+	depositContract := env.Devnet.Bindings.BindingsL1.OptimismPortal
+	depositor := l1Vault.CreateAccount(env.TimeoutCtx(time.Minute), l1, big.NewInt(3*params.Ether))
+	l2Vault.InsertKey(l1Vault.FindKey(depositor))
+	l2Opts := l2Vault.KeyedTransactor(depositor)
+
+	// Fund account on L2
+	doDeposit(t, env, depositor, big.NewInt(0.5*params.Ether), false, nil)
+
+	// Deploy the failure contract on L2
+
+	_, deployTx, failureContract, err := hivebindings.DeployFailure(l2Opts, l2)
+	require.NoError(t, err)
+	_, err = optimism.WaitReceipt(env.TimeoutCtx(30*time.Second), l2, deployTx.Hash())
 	require.NoError(t, err)
 
+	// Create the revert() call
+	l2Opts.NoSend = true
+	l2Opts.GasLimit = 1_000_000
+	revertTx, err := failureContract.Fail(l2Opts)
+	require.NoError(t, err)
+
+	// Create garbage data
+	randData := make([]byte, 32)
+	_, err = rand.Read(randData)
+	require.NoError(t, err)
+
+	testData := [][]byte{
+		randData,
+		revertTx.Data(),
+	}
+	mintAmount := big.NewInt(0.5 * params.Ether)
```

Comment:
I know it's an untyped constant and produces a non-zero value after multiplication, but it still sketches me out to use floating point math in code that uses integers. We don't want a 0 mint amount.

Reply:
I can make it into a constant that uses int math.

Reply:
Will do that as a separate PR though, this is used all over the place.

Reply:
Tracked in ENG-2861.